src/interface/swarm_comms.cpp

```diff
@@ -26,265 +26,105 @@
+                                               : bsize.xmin_[2] + (k - 0.5) * dx_test[2]};
+
+        // Account for periodic boundary conditions by applying BCs to x_test
+        // Assumes mesh is hyper-rectangular and Mesh::mesh_size represents the entire
+        // domain.
+        auto &msize = pmb->pmy_mesh->mesh_size;
+        if (pmb->boundary_flag[0] == BoundaryFlag::periodic) {
+          if (x_test[0] < msize.xmin(X1DIR)) {
+            x_test[0] = msize.xmax(X1DIR) - (msize.xmin(X1DIR) - x_test[0]);
+          }
         }
-      } else if (j == 1) {
-        if (k == -1) {
-          neighbor_indices_h(0, 3, 1) = n;
-          neighbor_indices_h(0, 3, 2) = n;
-        } else if (k == 0) {
-          neighbor_indices_h(1, 3, 1) = n;
-          neighbor_indices_h(1, 3, 2) = n;
-          neighbor_indices_h(2, 3, 1) = n;
-          neighbor_indices_h(2, 3, 2) = n;
-        } else if (k == 1) {
-          neighbor_indices_h(3, 3, 1) = n;
-          neighbor_indices_h(3, 3, 2) = n;
+        if (pmb->boundary_flag[1] == BoundaryFlag::periodic) {
+          if (x_test[0] > msize.xmax(X1DIR)) {
+            x_test[0] = msize.xmin(X1DIR) + (x_test[0] - msize.xmax(X1DIR));
+          }
         }
-      }
-    } else if (i == 1) {
-      if (j == -1) {
-        if (k == -1) {
-          neighbor_indices_h(0, 0, 3) = n;
-        } else if (k == 0) {
-          neighbor_indices_h(1, 0, 3) = n;
-          neighbor_indices_h(2, 0, 3) = n;
-        } else if (k == 1) {
-          neighbor_indices_h(3, 0, 3) = n;
+        if (ndim > 1) {
+          if (pmb->boundary_flag[2] == BoundaryFlag::periodic) {
+            if (x_test[1] < msize.xmin(X2DIR)) {
+              x_test[1] = msize.xmax(X2DIR) - (msize.xmin(X2DIR) - x_test[1]);
+            }
+          }
+          if (pmb->boundary_flag[3] == BoundaryFlag::periodic) {
+            if (x_test[1] > msize.xmax(X2DIR)) {
+              x_test[1] = msize.xmin(X2DIR) + (x_test[1] - msize.xmax(X2DIR));
+            }
+          }
         }
-      } else if (j == 0) {
-        if (k == -1) {
-          neighbor_indices_h(0, 1, 3) = n;
-          neighbor_indices_h(0, 2, 3) = n;
-        } else if (k == 0) {
-          neighbor_indices_h(1, 1, 3) = n;
-          neighbor_indices_h(1, 2, 3) = n;
-          neighbor_indices_h(2, 1, 3) = n;
-          neighbor_indices_h(2, 2, 3) = n;
-        } else if (k == 1) {
-          neighbor_indices_h(3, 1, 3) = n;
-          neighbor_indices_h(3, 2, 3) = n;
+        if (ndim > 2) {
+          if (pmb->boundary_flag[4] == BoundaryFlag::periodic) {
+            if (x_test[2] < msize.xmin(X3DIR)) {
+              x_test[2] = msize.xmax(X3DIR) - (msize.xmin(X3DIR) - x_test[2]);
+            }
+          }
+          if (pmb->boundary_flag[5] == BoundaryFlag::periodic) {
+            if (x_test[2] > msize.xmax(X3DIR)) {
+              x_test[2] = msize.xmin(X3DIR) + (x_test[2] - msize.xmax(X3DIR));
+            }
+          }
```

Comment:
@lroberts36 We talked about a method of maximally refining the current block and all neighbor blocks and comparing all refined blocks to accommodate periodic boundary conditions, but because `mesh_size` is the entire extent of the hyperrectangular domain, just applying boundary conditions to the test point works fine (I tested downstream with a periodic BC across two trees). I think this method also results in fewer comparison operations since we are comparing each ghost halo test block with each neighbor, rather than each ghost halo test block with each daughter of each neighbor. 

So I'm inclined to suggest this method get merged in because it's simple, supports the general hyperrectangular case, and is already done. We have a near-term need for SMR downstream. But if you think this is really dangerous in terms of future proofing lets discuss more.